### PR TITLE
feat: implement issuer and certificaterequest api and controllers

### PR DESCRIPTION
**Pull request by mzeevi**
This PR includes all the code needed for the external issuer.

### CODE REVIEW

File: internal/setup/controllers.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+	return namespace, err
+}
+
+// getInClusterNamespace returns the name of the namespace where the pod is currently running
+// Copied from controller-runtime/pkg/leaderelection.
+func getInClusterNamespace() (string, error) {
+	_, err := os.Stat(inClusterNamespacePath)
+	if os.IsNotExist(err) {
+		return "", errors.New("not running in-cluster")
```
Comment (arielsepton) [inline internal/setup/controllers.go:85]:
```suggestion
		return "", errNotInCluster
```

**Review by arielsepton — CHANGES_REQUESTED** [review]
Everything looks excellent. I would suggest adding tests for `internal/issuer/validate/validate.go`, but I trust your judgment on this. I've left a few minor comments.

**Comment by mzeevi** [comment]
Thanks for the review, I've pushed the changes. As for the missing tests, I'll open issues with some stuff that need to be done so others can get into this code.

**Review by arielsepton — APPROVED** [review]
/lgtm

**Pull request merged**